### ISSUE TITLE
Training problem

### ISSUE THREAD

**Issue by ray823** [sagieppel/Fully-convolutional-neural-network-FCN-for-semantic-segmentation-Tensorflow-implementation#14]
Excuse me, I have some questions to ask. Can your fcn model predict multiple categories from a single image?
For example, to train a model to recognize dogs and cats, I can manually label a single image to generate a mask image, and the image has two categories of dogs and cats at the same time, and I don't need to divide dogs and cats into two categories Training, is this possible?
Just like the prediction result of your cover, liquid and glass should not be trained separately, and his mask image should be a mixture of two types in the same mask image and not binary classification

**Comment by sagieppel** [comment]
Yes, but this specific model is old, I would recommend you have a look at this instead:
[https://towardsdatascience.com/train-neural-net-for-semantic-segmentation-with-pytorch-in-50-lines-of-code-830c71a6544f](https://towardsdatascience.com/train-neural-net-for-semantic-segmentation-with-pytorch-in-50-lines-of-code-830c71a6544f
)
